TA_dataset/bin/dataset_search_command.py

```diff
@@ -436,6 +443,13 @@ def generate(self):
                         "DataSetFunction=makeRequest, destination={}, startTime={}"
                         .format(ds_url, time.time())
                     )
+                    tenant_related_payload = get_tenant_related_payload(
```

Comment:
Why not to call `get_tenant_related_payload` only once somewhen before the `try` block and then pass only `tenant_related_payload` to all functions above?

Reply:
Done